client/src/main/java/org/apache/uniffle/client/impl/ShuffleWriteClientImpl.java

```diff
@@ -914,7 +915,16 @@ public void sendAppHeartbeat(String appId, long timeoutMs) {
               LOG.warn("Failed to send heartbeat to " + shuffleServerInfo);
             }
           } catch (Exception e) {
-            LOG.warn("Error happened when send heartbeat to " + shuffleServerInfo, e);
+            if (closed) {
+              // this client has been closed, so failing heartbeat calls are expected
+              LOG.info(
+                  "Error happened when send heartbeat to "
+                      + shuffleServerInfo
+                      + " after shuffle write client has been closed",
+                  e);
+            } else {
+              LOG.warn("Error happened when send heartbeat to " + shuffleServerInfo, e);
```

Comment:
Ditto.

Reply:
For other similar logging issues in this PR, please modify them together as well. :)

Reply:
There are still a lot of places using `+ shuffleServerInfo`, you can search for them.